aws/data_source_aws_ecr_repository.go

```diff
@@ -54,17 +55,22 @@ func dataSourceAwsEcrRepositoryRead(d *schema.ResourceData, meta interface{}) er
 	}
 
 	repository := out.Repositories[0]
-
-	log.Printf("[DEBUG] Received ECR repository %s", out)
+	arn := aws.StringValue(repository.RepositoryArn)
 
 	d.SetId(aws.StringValue(repository.RepositoryName))
-	d.Set("arn", repository.RepositoryArn)
+	d.Set("arn", arn)
 	d.Set("registry_id", repository.RegistryId)
```

Comment:
Do any of these need `aws.StringValue`s as well?

Reply:
Good question and no -- `ResourceData.Set()` automatically handles pointers: https://github.com/hashicorp/terraform-plugin-sdk/blob/6f51fdc99e1e3a27b7445f873f217e2d961869a3/helper/schema/resource_data.go#L178